### PR TITLE
Fixed the bug of log depth calculation error when using coroutine such as swoole.

### PR DESCRIPTION
**Pull request by limingxinleo**
https://github.com/hyperf/hyperf/issues/4740

https://github.com/hyperf/hyperf/pull/4754

### CODE REVIEW

**Comment by limingxinleo** [comment]
Then I just need to rewrite `logDepth` to solve this problem easily

File: src/Monolog/Logger.php [hunk 1]
```diff
@@ -148,9 +148,9 @@ class Logger implements LoggerInterface, ResettableInterface
     protected $exceptionHandler;
 
     /**
-     * @var int Keeps track of depth to prevent infinite logging loops
+     * @var LogDepthInterface Keeps track of depth to prevent infinite logging loops
      */
-    private $logDepth = 0;
+    protected $logDepth;
```
Comment (stof) [inline src/Monolog/Logger.php:153]: why protected ?

Comment (limingxinleo) [reply src/Monolog/Logger.php:153]: Because I will create a Logger which extends Monolog

And then rewrite logDepth

**Comment by limingxinleo** [comment]
> I don't see how storing the depth in a separate class solves the bug.

I can rewrite logDepth like this

class LogDepth
{
public function increment()｛return 0}
public function decrement() ｛
return 0}
}

File: src/Monolog/Logger.php (hunk 1 above)
Comment (stof) [reply src/Monolog/Logger.php:153]: Well, if you want to replace the tracking of the log depth, what you need is an interface for that object (to allow providing another implementation) and a way to inject a custom implementation.

But inheritance-based extension points are the worse ones both in term of maintenance (it is a lot harder to write a BC layer for a protected hooking point than for a clean extension point) and in term of flexibility (if 2 packages extend the Logger to hook into different places, you have to choose between them, you cannot combine their extensions)

**Comment by stof** [comment]
> I can rewrite logDepth like this

That would just disable the tracking of logging loops entirely, which could then be achieved by passing a boolean flag to control that tracking in a much simpler way.

**Comment by limingxinleo** [comment]
> That would just disable the tracking of logging loops entirely, which could then be achieved by passing a boolean flag to control that tracking in a much simpler way.

You are right

**Pull request closed without merging**